plugins/org.yakindu.sct.doc.user/src/user-guide/sctunit.textile

```diff
@@ -5,370 +5,941 @@ p.
 
 ==<!-- Start sctunit_keyword_assert -->==
 
-h4. assert
+h4(#sctunit_assertions). Assertions
+
+When it comes to testing, the most important statement is the _assertion_. It evaluates a condition that must be fulfilled for the test to not fail.
+
+==<div class="example">==
+
+Example 1:
+
+bc. assert sum == 42
 
-The expression followed by the 'assert' keyword expects a boolean condition to be true, the test will fail otherwise. It is possible to add an optional message to the assertion statement to clarify the expressions intention. The following example shows a statement that asserts if a state is 'active' or not.
+This statement asserts that the variable _sum_ has a value of 42. If this is the case, the operation continues. If not, the test fails and is discontinued.
 
-bc(prettyprint). assert active(MyStatechart.regionName.StateName) "Expected StateName to be active"
-assert !active(MyStatechart.regionName.AnotherStateName)
+==</div>==
+
+==<div class="example">==
+
+Example 2:
+
+bc. assert active (myStatechart.main_region.State_A)
+assert !active (myStatechart.main_region.State_B) message "State_B must not be active here."
+
+The first statement asserts that state _State_A_ in region _main_region_ in statechart _myStatechart_ is active. If this is not the case, the test fails.
 
-It is also possible to assert variable values. You can use any boolean comparator and its also possible to access variables in Interfaces and values of events:
+The second statement asserts that _State_B_ is not active. Otherwise the test fails with the error message "State_B must not be active here.". 
 
-bc(prettyprint). assert myVariable1 == 42
-assert myVariable2 == "Sense of life"
-assert myVariable3 => 41
-assert InterfaceName.MyEvent == 5
+bq. *Please note:* "active(…)":../user-guide/statechart_language#active-state is a built-in function of the statechart language.
+
+==</div>==
+
+Generally, an assertion consists of the keyword @assert@, followed by a boolean expression, optionally followed by the keyword @message@ and an error message text. The assertion expects the boolean expression to be _true_ to continue the test. If it evaluates to _false_ the test fails. The optional message can be used to clarify what went wrong.
 
-p. ==<!-- End sctunit_keyword_assert -->==
 ==<!-- Start sctunit_keyword_called -->==
 
-h4. called
+h5. Asserting an operation call
 
-The expression followed by the 'called' keyword expects a reference to an operation that will be called and optional the keyword 'with' and the parameters of the call. If the referenced operation wasn't called or wasn't called with the right parameters, the test will fail. The following example shows a statement that asserts if a operation was called or not.
+A special variant of the assertion uses the _called_ expression. It checks whether a certain operation has been called (executed), typically by some action in the statechart.
```

Comment:
Hm, I rephrased the wording somewhat, but basically I'd like to keep it as it is.

The underlying general principle is that **people learn by three examples and the theory, not vice versa.**